### PR TITLE
extract  suggest plugin

### PR DESCRIPTION
**Pull request by mixmix**
Talked with @mmckegg about wanting to use patchworks suggest mentioning system
Extracts the suggestion of profiles into its own file + plufing.

API is now  `ssb.patchwork.suggest.profile` (leaving space for `suggest.meme` etc in the future!)

Personally this feels like a really nice clean separation of concerns

### CODE REVIEW

File: sbot/suggest.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          // add following attribute
+          result = result.map(x => extend(x, { following: following.has(x.id) }))
+
+          cb(null, result)
+        } else if (defaultIds && defaultIds.length) {
+          cb(null, defaultIds.map(id => suggestCache[id]))
+        } else {
+          let ids = recentFriends.slice(-(limit || 20)).reverse()
+          let result = ids.map(id => suggestCache[id])
+          result = result.map(x => extend(x, { following: following.has(x.id) }))
```
Comment (christianbundy) [inline sbot/suggest.js]: It looks like the `ssb.ebt.block` is magically resolved, but now I'm seeing:

```
{ message: 'Cannot read property \'id\' of undefined',
  name: 'TypeError',
  stack: 'TypeError: Cannot read property \'id\' of undefined\n    at result.map.x (/home/christianbundy/src/ssbc/patchwork/sbot/suggest.js:132:75)\n    at Array.map (<anonymous>)\n    at /home/christianbundy/src/ssbc/patchwork/sbot/suggest.js:132:27\n    at update (/home/christianbundy/src/ssbc/patchwork/sbot/suggest.js:107:7)\n    at Object.profile (/home/christianbundy/src/ssbc/patchwork/sbot/suggest.js:114:7)\n    at Object.hooked (/home/christianbundy/src/ssbc/patchwork/node_modules/ssb-server/node_modules/hoox/index.js:10:15)\n    at Object.localCall (/home/christianbundy/src/ssbc/patchwork/node_modules/ssb-server/node_modules/muxrpc/local-api.js:31:29)\n    at Object.<anonymous> (/home/christianbundy/src/ssbc/patchwork/node_modules/ssb-server/node_modules/muxrpc/local-api.js:37:22)\n    at Object.request (/home/christianbundy/src/ssbc/patchwork/node_modules/ssb-server/node_modules/muxrpc/stream.js:48:17)\n    at PacketStream._onrequest (/home/christianbundy/src/ssbc/patchwork/node_modules/ssb-server/node_modules/packet-stream/index.js:161:17)' }
```

@mixmix any chance that's resolved in ssb-suggest? :upside_down_face:

Comment (Powersource) [reply sbot/suggest.js]: @christianbundy I just tried switching to ssb-suggest (first commit) in https://github.com/ssbc/patchwork/tree/use-ssb-suggest and things work without errors. (gonna try to get ssb-meme in there as well before making a PR though :D )

**Comment by christianbundy** [comment]
Just merged ssb-suggest in e4cd438cfab8c883905554155c781c7aafbeb5c4, closing!

(please re-open if this was a mistake!)

**Comment by Powersource** [comment]
This PR also removed the user-suggesting-related code from sbot/profile.js. I think that's still in master now?

(reopening to not forget to discuss)

**Comment by christianbundy** [comment]
Nice catch, thanks! I'm going to merge `master` into this branch, then I think we can remove `sbot/suggest.js` in favor of `ssb-suggest`?

**Comment by Powersource** [comment]
Sounds good!

**Comment by christianbundy** [comment]
@Powersource 

Sweet, just  pushed a commit and it's now looking great on my end. Please let me know if you notice anything funky or feel free to merge. :+1:

**Comment by Powersource** [comment]
I could have sworn typing `@kas` suggested more than 2 of his accounts when using ssb-suggest, but I can't replicate that even when checkout'ing `e4cd438cfab8c883905554155c781c7aafbeb5c4`. Could be that ssb-suggest got updated, it's gotten a bunch of new commits since this PR was opened.

Stuff seems to work so merging.

**Pull request merged**